TombEngine/Scripting/Internal/TEN/Effects/EffectsFunctions.cpp

```diff
@@ -582,6 +638,80 @@ namespace TEN::Scripting::Effects
 			convertedEdgeFeatherID, /*convertedLengthFeatherID, */convertedBlendID);
 	}
 
+	/// Emit a visual sink effect.
+	// @function EmitSink
+	// @tparam Vec3 pos World position.
+	// @tparam Vec3 dir Directional velocity of the particles in world units per second.
+	// @tparam[opt=512] float radius Radius of emitter. The particles will be emitted inside the circle of provided radius measured from centre of world position.
+	// @tparam[opt=1] float life Lifespan in seconds.
+	// @tparam[opt=0] float friction Friction affecting velocity over time in world units per second.
+	// @tparam[opt=25] float maxSize Max size of the particle.
+	// @tparam[opt=Color(128&#44; 128&#44; 128)] Color startColor Color at start of life.
+	// @tparam[opt=Color(0&#44; 0&#44; 0)] Color endColor Color at end of life.
+	// @tparam[opt=Objects.ObjID.DEFAULT_SPRITES] Objects.ObjID.SpriteConstants spriteSeqID Sprite sequence slot ID.
+	// @tparam[opt=14 (UNDERWATER_DUST)] int spriteID Sprite ID in the sprite sequence slot.
+	static void EmitSink(const Vec3& pos, const Vec3& dir, TypeOrNil<float> radius, TypeOrNil<float> life, TypeOrNil<float> friction, TypeOrNil<float> maxSize, TypeOrNil<ScriptColor> startColor, TypeOrNil<ScriptColor> endColor, TypeOrNil<GAME_OBJECT_ID> spriteSeqID, TypeOrNil<int> spriteID)
```

Comment:
In my opinion, `EmitSink` is a little misleading name. Historically, "sink" in TRLE community means "an object towards which Lara is moving when used together with sink trigger". Therefore, some people may falsely think that this function will physically move Lara in the same way sink trigger does. I suggest renaming it to something more generic, especially because this effect may be called not only underwater.